cloudflare/resource_cloudflare_zone_dnssec.go

```diff
@@ -0,0 +1,156 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// The supported status for a Zone DNSSEC setting
+const (
+	DNSSECStatusActive   = "active"
+	DNSSECStatusDisabled = "disabled"
+)
+
+func resourceCloudflareZoneDNSSEC() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareZoneDNSSECCreate,
+		Read:   resourceCloudflareZoneDNSSECRead,
+		Update: resourceCloudflareZoneDNSSECUpdate,
+		Delete: resourceCloudflareZoneDNSSECDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"zoneid": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: zoneDiffFunc,
+			},
+			"status": {
```

Comment:
As this is really the only value that we can control, it seems like this should be `Required: true`. Any reason why it isn't at the moment?

Reply:
I see below you're setting the values which makes a bit of sense here. I'm not opposed to it but still on the fence whether we should be accepting the value from the end user and passing it through instead.

Reply:
`Required: true` is already set.

Reply:
@jacobbednarz 

https://github.com/cloudflare/terraform-provider-cloudflare/pull/852#discussion_r517648569 okay, I'll make that change. I was always getting an error message with the `DELETE` API, so I thought it would be better to disable DNSSEC on delete.

```bash
curl -X DELETE "https://api.cloudflare.com/client/v4/zones/${CLOUDFLARE_ZONE_ID}/dnssec" -H "Authorization: Bearer ${CLOUDFLARE_API_TOKEN}"  -H "Content-Type:application/json" | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   124    0   124    0     0    122      0 --:--:--  0:00:01 --:--:--   122
{
  "success": false,
  "errors": [
    {
      "code": 9109,
      "message": "Unauthorized to access requested resource"
    }
  ],
  "messages": [],
  "result": null
}
```
It could be that the domain registrar is not CloudFlare. I raised a support ticket and didn't get a response back. Do you think I should make a new variable to handle this situation?

Reply:
I'm sorry, I don't follow. You have mentioned a couple of things here which seem unrelated to the initial comment. Can you explain a little more?

Reply:
Sorry for the confusion. As per https://github.com/cloudflare/terraform-provider-cloudflare/pull/852#discussion_r517648569 I assumed you'd wanted the `status` to be a user controllable parameter. I was explaining the reason behind not exposing it and setting it to `disabled` on `DELETE`. Hope this clarifies the confusion :D